mesonbuild/coredata.py

```diff
@@ -943,17 +962,19 @@ class BuiltinOption(Generic[_T, _U]):
     Currently doesn't support UserIntegerOption, or a few other cases.
     """
 
-    def __init__(self, opt_type: Type[_U], description: str, default: Any, yielding: Optional[bool] = None, *,
+    def __init__(self, opt_type: Type[_U], description: str, default: Any, yielding: Optional[bool] = True, *,
                  choices: Any = None):
         self.opt_type = opt_type
         self.description = description
         self.default = default
         self.choices = choices
         self.yielding = yielding
 
-    def init_option(self) -> _U:
+    def init_option(self, value=None) -> _U:
```

Comment:
It would be nice to have a type annotation for value

Reply:
Can be any type, depending on which UserOption subclass it is. value is not annotated in other places.

Reply:
@dcbaker I'm not familiar with annotation system, what should I put here?